executor/executor.go

```diff
@@ -138,9 +138,11 @@ func doRunSetupCommand(emitter screwdriver.Emitter, f *os.File, r io.Reader, set
 	return nil
 }
 
-func doRunCommand(guid, path string, emitter screwdriver.Emitter, f *os.File, fReader io.Reader) (int, error) {
+func doRunCommand(guid, path string, emitter screwdriver.Emitter, f *os.File, fReader io.Reader, stepName string) (int, error) {
 	executionCommand := []string{
 		"export SD_STEP_ID=" + guid,
+		// escape not necessary because step name is limited to [A-Za-z0-9_-]
+		";export SD_STEP_NAME=" + stepName,
 		";. " + path,
 		";echo",
```

Comment:
Not related to my change, but I noticed that the commit introducing this 'echo' statement (2556c791 / #109) unintentionally broke the status check on the next line (`$?` comes from `echo` rather than from `source`).

It shouldn't really make a difference (i.e. `echo guid $?` would be skipped because of `set -e` if the user code results in anything other than success). But it's not the original intent of the code (and confusing). It should either just be `echo 0` (because we know the line will only ever be reached on success) or something like `"; printf \\\\n " + guid + " $?\n"` and the extraneous `echo` removed.

If you have a preferred fix among those options, I am happy to send a separate pr.

Reply:
sure, u can use that printf or echo, whichever can wrap a newline before and after will do